### PR TITLE
Update Expr.hx API docs

### PR DESCRIPTION
**Pull request by markknol**
* some bits about names
* use vars on Catch/Var and MetaDataEntry, to be able to display the api docs HaxeFoundation/dox#205
* added some enum options to Field members, for convenience while coding.
* fix Constant regular expression example list (add linebreak)
* additional info about Position

### CODE REVIEW

File: std/haxe/macro/Expr.hx [hunk 1]
```diff
@@ -26,7 +26,11 @@ extern enum Position {
 }
 #else
 /**
-	Represents a position in a file.
+	Represents the position of where a given type, field, function or variable
+	is defined. This is used by the compiler to display for example errors 
+	or to provide code completion information.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: I don't like the listing here, it's not exhaustive and maintaining it to be exhaustive would be annoying and pointless.

File: std/haxe/macro/Expr.hx [hunk 2]
```diff
@@ -296,18 +301,22 @@ typedef Case = {
 typedef Var = {
 	/**
 		The name of the variable.
+		
+		Haxe variable names start with an underscore `_`, a dollar `$`, a lower-case
+		character `a-z` or an upper-case character `A-Z`. After that, any combination 
+		and number of `_`, `A-Z`, `a-z` and `0-9` may follow.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: This information suggests that the naming is somehow enforced. I think the only check that is done after parsing is the leading dollar, so this is somewhat misleading.

It _might_ be useful to mention that this is only guaranteed to work if the name conforms to the Haxe identifier rules.

File: std/haxe/macro/Expr.hx [hunk 3]
```diff
@@ -539,6 +548,9 @@ enum ComplexType {
 
 /**
 	Represents a type path in the AST.
+
+	Haxe type paths can only contain underscores `_` or lower-case
+	characters `a-z`.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: That's not what TypePath is as it contains the type name and also type parameters, which makes this information untrue.

File: std/haxe/macro/Expr.hx [hunk 4]
```diff
@@ -602,7 +614,7 @@ typedef TypeParamDecl = {
 	@:optional var params : Array<TypeParamDecl>;
 
 	/**
-		The metadata of the type parameter.
+		The metadata of the type parameter, which is an array of `MetadataEntry`.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: I mean yes, it is, but why mention that explicitly here? We don't do that in any other place...

File: std/haxe/macro/Expr.hx [hunk 5]
```diff
@@ -693,23 +714,43 @@ typedef Metadata = Array<MetadataEntry>;
 typedef Field = {
 	/**
 		The name of the field.
+		
+		Haxe field names start with an underscore `_`, a dollar `$`, a lower-case
+		character `a-z` or an upper-case character `A-Z`. After that, any combination 
+		and number of `_`, `A-Z`, `a-z` and `0-9` may follow.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: See above

File: std/haxe/macro/Expr.hx [hunk 6]
```diff
@@ -693,23 +714,43 @@ typedef Metadata = Array<MetadataEntry>;
...
 		The access modifiers of the field. By default fields have private access.
+		
+		 * `Access.APublic` Public access modifier, grants access from anywhere.
+		 * `Access.APrivate` Private access modifier, grants access to class and its 
+		    sub-classes only.
+		 * `Access.AStatic` Static access modifier.
+		 * `Access.AOverride` Override access modifier.
+		 * `Access.ADynamic` Dynamic (re-)bindable access modifier.
+		 * `Access.AInline` Inline access modifier. Allows expressions to be directly 
+		    inserted inplace of calls to them.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: "calls" is not good because we also have inline variables.

File: std/haxe/macro/Expr.hx [hunk 7]
```diff
@@ -693,23 +714,43 @@ typedef Metadata = Array<MetadataEntry>;
...
 	**/
 	@:optional var access : Array<Access>;
 
 	/**
-		The kind of the field.
+		The kind is the type of the field. 
+		
+		 * `FieldType.FVar` for a variable.
+		 * `FieldType.FFun` for a function.
+		 * `FieldType.FProp` for a property with getter and setter.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: Not accurate, it doesn't necessarily have to have a getter or setter.

File: std/haxe/macro/Expr.hx (hunk 6 above)
Comment (Simn) [inline std/haxe/macro/Expr.hx]: Also, why list all this here if we document the constructors on the type anyway?

File: std/haxe/macro/Expr.hx [hunk 8]
```diff
@@ -805,6 +851,10 @@ typedef TypeDefinition = {
 
 	/**
 		The name of the type definition.
+		
+		Haxe type names start with an underscore `_` or an upper-case
+		character `A-Z`. After that, any combination and number of `_`, `A-Z`, `a-z` 
+		and `0-9` may follow.
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: See above

**Comment by markknol** [comment]
Lets close this. I thought it would add something but if it gives confusion lets not waste time here. Ill do a new request later with the updated typedefs for the api docs.

**Pull request closed without merging**